src/App.tsx

```diff
@@ -1,24 +1,30 @@
 import { ThemeProvider } from "@mui/material";
-import React from "react";
+import * as React from "react";
 import { BrowserRouter as Router } from "react-router-dom";
 import NavBar from "./NavBar";
 import { AppRoutes } from "./AppRoutes";
 import { muiTheme, theme } from "./components/theme/theme";
+import { UserContext } from "shared/UserContext";
+import { useLogin } from "shared/hooks/useLogin";
 
 export default function App() {
+  const { user, loginFunction, logOutFunction } = useLogin();
+
   return (
     <div
       style={{
         backgroundColor: theme.colors.background,
         color: theme.colors.textPrimary,
       }}
     >
-      <ThemeProvider theme={muiTheme}>
-        <Router>
-          <NavBar />
-          <AppRoutes />
-        </Router>
-      </ThemeProvider>
+      <UserContext.Provider value={{ user, loginFunction, logOutFunction }}>
```

Comment:
This provider makes sure that any component inside it can access the values provided in `value`. This is how I made sure that all files in the app can now access the user's info.